java/shared/lang/user_input.yml

```diff
@@ -0,0 +1,57 @@
+type: shared
+languages:
+  - java
+patterns:
+  - pattern: $<REQUEST>.$<REQUEST_METHOD>();
+    filters:
+      - variable: REQUEST
+        detection: java_shared_lang_parameter_request
+      - variable: REQUEST_METHOD
+        values:
+          - getCookies
+          - getHeader
+          - getRequestURI
+          - getRequestURL
+          - getAttribute
+          - getInputStream
+          - getQueryString
+          - getParameter
+          - getParameterMap
+          - getParameterNames
+          - getParameterValues
+          - getReader
+          - getHeaderNames
+          - getPart
+          - getParts
+  - pattern: $<SERVLET_REQUEST>.getHeader($<HEADER>)
+    filters:
+    - variable: SERVLET_REQUEST
+      detection: java_shared_lang_parameter_request
+    - variable: HEADER
+      string_regex: \A(?i)(Host|Referer|User-Agent)\z
```

Comment:
Not sure this should be part of the shared rule
If anything, I think it should be its own shared rule. WDYT?

Reply:
Where would we use it, if it's a shared rule?

Reply:
It looks like we're matching any `getHeader` call in the previous pattern anyway. The user can set the value of any of them anyway, so that makes sense to me. Should we remove this pattern?